daly-bms-uart.cpp

```diff
@@ -165,7 +380,7 @@ bool Daly_BMS_UART::validateChecksum()
     }
 
 #ifdef DALY_BMS_DEBUG
-    Serial.printf("<DALY-BMS DEBUG> Calculated checksum: 0x%x, Received checksum: 0x%x\n", checksum, this->my_rxBuffer[XFER_BUFFER_LENGTH - 1]);
+    DEBUG_SERIAL.print("<DALY-BMS DEBUG> Calculated checksum: 0x" + (String)checksum + ", Received checksum: 0x" + (String)this->my_rxBuffer[XFER_BUFFER_LENGTH - 1] + "\n");
```

Comment:
If you cast `checksum` to a string it will print out in the integer representation not the hex representation.
```suggestion
    DEBUG_SERIAL.print("<DALY-BMS DEBUG> Calculated checksum: " + (String)checksum + ", Received checksum: " + (String)this->my_rxBuffer[XFER_BUFFER_LENGTH - 1] + "\n");
```

Reply:
oh, yes